host/class/uac/usb_host_uac/include/usb/uac_host.h

```diff
@@ -210,6 +212,28 @@ esp_err_t uac_host_uninstall(void);
  */
 esp_err_t uac_host_device_open(const uac_host_device_config_t *config, uac_host_device_handle_t *uac_dev_handle);
 
+/**
+ * @brief Open a UAC logic device/interface with specific VID, PID and interface number
+ *
+ * @note The config->addr is not used in this function, the VID , PID and config->iface_num should be provided.
+ *       If multiple hardware devices with the same VID and PID are connected through USB hub, the first one will be opened.
```

Comment:
Also, you are mentioning an external hub. I tried running the test app, with an UAC device connected to the ESP through an external hub, but the test app couldn't find the UAC device. It keeps finding only PID VID of the ext. hub.

`D (7762) uac-host: Found Device VID 0x05E3, PID 0x0608`
`W (7812) uac-host: Wait for device VID 0x1395, PID 0x0025 timeout`

Without the external hub, the UAC device is found correctly.

Reply:
@peter-marcisovsky Thanks for your test, I will check it

Reply:
@peter-marcisovsky I got the same test result as you, no downstream device was detected and the `usbh_dev_addr_list_fill` always returns one device. The log is below. @roma-jam would you please point out the reason? 

```
I (4941) UAC_TEST: USB Host installed
D (4941) uac-host: USB UAC handling start
I (4941) uac-host: Install Succeed, Version: 1.1.0
I (4944) UAC_TEST: UAC Class Driver installed
I (4949) UAC_TEST: Device not found, please connect the device
D (5251) HUB: Root port reset
D (5251) HUB: New device tree node (uid=1)
D (5251) ENUM: [0:0] Start processing, device address 0
D (5252) ENUM: [0:0] GET_SHORT_DEV_DESC OK
D (5256) USBH: Processing actions 0x4
D (5260) USBH: Default pipe device 0
D (5264) ENUM: [0:0] CHECK_SHORT_DEV_DESC OK
D (5328) ENUM: [0:0] SECOND_RESET OK
D (5328) ENUM: [0:0] SET_ADDR OK
D (5328) USBH: Processing actions 0x4
D (5328) USBH: Default pipe device 0
D (5331) ENUM: Assign address (dev_addr=1)
D (5335) ENUM: [0:0] CHECK_ADDR OK
D (5348) ENUM: [0:0] SET_ADDR_RECOVERY OK
D (5348) ENUM: [0:0] GET_FULL_DEV_DESC OK
D (5348) USBH: Processing actions 0x4
D (5350) USBH: Default pipe device 1
D (5354) ENUM: [0:0] CHECK_FULL_DEV_DESC OK
D (5358) ENUM: Selected bConfigurationValue=1
D (5363) ENUM: [0:0] SELECT_CONFIG OK
D (5366) ENUM: [0:0] GET_SHORT_CONFIG_DESC OK
D (5371) USBH: Processing actions 0x4
D (5375) USBH: Default pipe device 1
D (5378) ENUM: [0:0] CHECK_SHORT_CONFIG_DESC OK
D (5383) ENUM: [0:0] GET_FULL_CONFIG_DESC OK
D (5387) USBH: Processing actions 0x4
D (5391) USBH: Default pipe device 1
D (5395) ENUM: [0:0] CHECK_FULL_CONFIG_DESC OK
D (5399) ENUM: [0:0] GET_SHORT_LANGID_TABLE OK
D (5404) USBH: Processing actions 0x4
D (5407) USBH: Default pipe device 1
D (5411) ENUM: [0:0] CHECK_SHORT_LANGID_TABLE OK
D (5416) ENUM: [0:0] GET_FULL_LANGID_TABLE OK
D (5420) USBH: Processing actions 0x4
D (5424) USBH: Default pipe device 1
D (5427) ENUM: [0:0] CHECK_FULL_LANGID_TABLE OK
D (5432) ENUM: [0:0] GET_SHORT_MANU_STR_DESC OK
D (5437) USBH: Processing actions 0x4
D (5440) USBH: Default pipe device 1
D (5444) ENUM: [0:0] CHECK_SHORT_MANU_STR_DESC OK
D (5449) ENUM: [0:0] GET_FULL_MANU_STR_DESC OK
D (5453) USBH: Processing actions 0x4
D (5457) USBH: Default pipe device 1
D (5461) ENUM: [0:0] CHECK_FULL_MANU_STR_DESC OK
D (5465) ENUM: [0:0] GET_SHORT_PROD_STR_DESC OK
D (5470) USBH: Processing actions 0x4
D (5474) USBH: Default pipe device 1
D (5477) ENUM: [0:0] CHECK_SHORT_PROD_STR_DESC OK
D (5482) ENUM: [0:0] GET_FULL_PROD_STR_DESC OK
D (5487) USBH: Processing actions 0x4
D (5490) USBH: Default pipe device 1
D (5494) ENUM: [0:0] CHECK_FULL_PROD_STR_DESC OK
D (5499) ENUM: String iSerialNumber not set, skip
D (5503) ENUM: [0:0] GET_SHORT_SER_STR_DESC OK
D (5508) ENUM: String iSerialNumber not set, skip
D (5513) ENUM: [0:0] CHECK_SHORT_SER_STR_DESC OK
D (5517) ENUM: String iSerialNumber not set, skip
D (5522) ENUM: [0:0] GET_FULL_SER_STR_DESC OK
D (5527) ENUM: String iSerialNumber not set, skip
D (5531) ENUM: [0:0] CHECK_FULL_SER_STR_DESC OK
D (5536) ENUM: [0:0] SET_CONFIG OK
D (5539) USBH: Processing actions 0x4
D (5543) USBH: Default pipe device 1
D (5547) ENUM: [0:0] CHECK_CONFIG OK
D (5551) ENUM: [0:0] Processing complete, new device address 1
D (5556) ENUM: [0:0] COMPLETE OK
D (5560) USBH: Processing actions 0x40
D (5564) USBH: New device 1
W (5567) uac-host: USB device with addr(1) is not UAC device
D (5955) uac-host: Found 1 devices
D (5955) uac-host: Found Device VID 0x14CD, PID 0x8601
I (5955) UAC_TEST: Device not found, please connect the device
D (6959) uac-host: Found 1 devices
D (6959) uac-host: Found Device VID 0x14CD, PID 0x8601
I (6959) UAC_TEST: Device not found, please connect the device
```

Reply:
@roma-jam  Test with msc example in master branh, through hub device also not be detected. 🤔

Reply:
Ok, It's clear to me now:

1. the current ESP-IDF master branch 8e4454b285ad39881c5bf3f440d8be617a2f18a8 currently only enum the Hub device, without enum the no downstream devices.
2. The hub test passed If based the internal gitlab branch `refactor/usb_host_examples_ext_hub_support` with final USB Hubs support

```
Running test uac device handling with known pid vid...
I (7158) UAC_TEST: USB Host installed
I (7158) uac-host: Install Succeed, Version: 1.1.0
I (7159) UAC_TEST: UAC Class Driver installed
I (7162) UAC_TEST: Device not found, please connect the device
W (7543) uac-host: USB device with addr(1) is not UAC device
I (8168) UAC_TEST: UAC Device opened: MIC
find UAC 1.0 Microphone device
interface number: 3
total alt interfaces number: 1
--------alt interface[1]--------- 
channels = 1 
bit_resolution = 16 
sample_freq: 
        8000
iManufacturer: G
iProduct: H
iSerialNumber: 2

...

```